### ISSUE TITLE
ValueError: num_samples should be a positive integer value, but got num_samples=0

### ISSUE THREAD

**Issue by seekingdeep** [NathanUA/U-2-Net#152]
@NathanUA @adakoda @vincentzhang @PDillis 
After fixing the [os bug](https://github.com/NathanUA/U-2-Net/pull/151/files), i am trying to train, but it is showing error:

```
(unet) home@home-lnx:~/programs/U-2-Net$ python u2net_train.py
/home/home/anaconda3/envs/unet/lib/python3.7/site-packages/torch/nn/_reduction.py:44: UserWarning: size_average and reduce args will be deprecated, please use reduction='mean' instead.
  warnings.warn(warning.format(ret))
---
train images:  0
train labels:  0
---
Traceback (most recent call last):
  File "u2net_train.py", line 93, in <module>
    salobj_dataloader = DataLoader(salobj_dataset, batch_size=batch_size_train, shuffle=True, num_workers=1)
  File "/home/home/anaconda3/envs/unet/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 262, in __init__
    sampler = RandomSampler(dataset, generator=generator)  # type: ignore
  File "/home/home/anaconda3/envs/unet/lib/python3.7/site-packages/torch/utils/data/sampler.py", line 104, in __init__
    "value, but got num_samples={}".format(self.num_samples))
ValueError: num_samples should be a positive integer value, but got num_samples=0
``

Have a look at my folder list:
```
home@home-lnx:~/programs/U-2-Net$ tree -d
.
├── model
├── saved_models
│   ├── face_detection_cv2
│   ├── u2net
│   └── u2netp
├── test_data
│   ├── test_images
│   ├── u2netp_results
│   └── u2net_results
└── train_data
    ├── DUTS-TE
    │   ├── DUTS-TE-Image
    │   └── DUTS-TE-Mask
    └── DUTS-TR
        ├── DUTS-TR-Image
        └── DUTS-TR-Mask
```

**Comment by NathanUA** [comment]
Please check the lines 50-52 in the u2net_train.py. You have to change that to match your dataset.

**Comment by seekingdeep** [comment]
My dataset is placed in the `train_data/DUTS-TR` folder, having the `DUTS-TR-Image` and `DUTS-TR-Mask`
so what should i change exactly in the `u2net_train.py`

**Comment by seekingdeep** [comment]
@NathanUA i am still getting same error, even after i modified the training script, have a look at my pull #154 

```
(unet) home@home-lnx:~/programs/U-2-Net$ python train.py --input_images ./train_data/image --input_masks ./train_data/mask --model_name u2net
/home/home/anaconda3/envs/unet/lib/python3.7/site-packages/torch/nn/_reduction.py:44: UserWarning: size_average and reduce args will be deprecated, please use reduction='mean' instead.
  warnings.warn(warning.format(ret))
---
train images:  0
train labels:  0
---
Traceback (most recent call last):
  File "train.py", line 104, in <module>
    salobj_dataloader = DataLoader(salobj_dataset, batch_size=batch_size_train, shuffle=True, num_workers=1)
  File "/home/home/anaconda3/envs/unet/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 262, in __init__
    sampler = RandomSampler(dataset, generator=generator)  # type: ignore
  File "/home/home/anaconda3/envs/unet/lib/python3.7/site-packages/torch/utils/data/sampler.py", line 104, in __init__
    "value, but got num_samples={}".format(self.num_samples))
ValueError: num_samples should be a positive integer value, but got num_samples=0
```

**Comment by seekingdeep** [comment]
solved @NathanUA  merge my latest pull #154
